src/main/java/CMV.java

```diff
@@ -150,7 +151,34 @@ private boolean LIC9() {
    * Computes the LIC 10 condition.
    */
   private boolean LIC10() {
-    // TODO: Implementation.
+    int E_PTS = parameters.E_PTS;
+    int F_PTS = parameters.F_PTS;
+    double AREA = parameters.AREA1;
+
+    if (points.length < 5) return false;
+    if (E_PTS < 1 || F_PTS < 1) return false;
+    if (E_PTS + F_PTS > points.length - 3) return false;
+
+    // I don't think the try/catch block is necessary logically, but put it there just in case...
```

Comment:
It really isn't, and the exception condition is fishy... It doesn't actually indicate that such a set doesn't exist.
You can prove this loop will never index out of the array:
The maximum value of `i` is `points.length - 3 - E_PTS - F_PTS - 1`
Due to the previous condition `if (E_PTS + F_PTS > points.length - 3) return false;`, this is sure to never underflow the index, so `points[i]` is safe.
Same goes for `points[i + E_PTS + 1]` since the computed index is in the range `0 < idx < points.length - 2 - F_PTS < points.length - 1`.
And last, `points[i + E_PTS + 1 + F_PTS + 1]`, the index is clearly larger than `0` and with the maximum value for `i` being `points.length - 3 - E_PTS - F_PTS - 1`, the computed index is at most `points.length - 2`.

Unless I've made a mistake somewhere I think I may have uncovered a bug where the last element is never looked at due to the loop bound. Can someone verify?

Reply:
IMO if we get to the catch block there is no such set. You are trying to access a point where E_PTS or F_PTS make you go outside the list, there is therefore no such set.

But the rest of your reasoning (that it won't index out of the array) I agree with, I made a similar mathematical deduction.

Reply:
I agree that you get `points.length - 2` instead of the desired `points.length - 1`. Since loops are exclusive at the top of their range, I assume we should have `-2` instead of `-3` in the for-condition to get the desired behaviour.